src/features/patients/pages/PatientForm.tsx

```diff
@@ -67,21 +95,26 @@ export const PatientForm: FC = () => {
   const { createPatient, createNameNote } = usePatientApi();
 
   const handleSubmit = () => {
-    if (
-      !(patientFormRef?.current?.isValid && patientFormRef?.current?.isValid)
-    ) {
-      return;
-    }
+    const patientValidator = jsonPatientSchemaValidator();
+    const surveyValidator = jsonPatientSurveySchemaValidator();
+    const patientIsValid = patientValidator(patientFormRef.current?.formData);
+    const surveyIsValid = surveyValidator(surveyFormRef.current?.formData);
 
-    setIsSubmitting(true);
-    createPatient(patientFormRef?.current?.formData || {})
-      .then(({ data }) =>
-        createNameNote(data.ID, patientEvent?.id || "", surveyFormRef?.current?.formData)
-      )
-      .then(() => {
-        setIsSubmitting(false);
-        setShowConfirmation(true);
-      });
+    if (surveyIsValid && patientIsValid) {
```

Comment:
I was also troubled by the lack of a prompt when validation fails, you click the button and the form just sits there. This is much better.. with the focus on the required field and a notification.